### ISSUE TITLE
[Suggestion] "Volumetric Connectors"

### ISSUE THREAD

**Issue by Xerma** [tsunamayo/Starship-EVO#3016]
Some speculation has been had in the Discord about how or if pressurization will work and how it might or might not work with ships made with bricks. One idea I had was to have something similar to what a Minecraft mod, Buildcraft, uses to customize the dimensions of a quarry.

Basically, this would be a new brick (no block version since normal pressurization would be block based anyways) where you can link them all together and it forms an area of pressure/air tightness. Now with you talking about how you might make it so shipyards and gravity units use a drag and stretch tool vs coordinates, that may be an even better alternative to this suggestion.

![20201001091204_1](https://user-images.githubusercontent.com/32974671/94889486-f84b4580-0441-11eb-8d3e-41d983bf930e.jpg)

**Comment by cloudrambler** [comment]
I made a similar suggestion #2590, though your idea of delimiting vertices with bricks is interesting.

Agree we need a method of including some bricks in the pressurization method otherwise how could we have functioning custom canopies etc.? I have tried building small ships with only blocks and cockpit modules and frankly they look and feel lifeless compared to a brick built ship.

Someone previously suggested using a special paint layer to protect blocks/bricks from being deleted. That method could also be used to select specific bricks to be included in the pressurization check, either as vertices or as a complete panel selection. I think such a technical paint layer, could be useful for many applications in Stevo.

**Comment by ProPeach** [comment]
So this is an interesting way of designating an area, but how does the game then check to see if the area is actually airtight, or if it has been breached? The fun part (and that bit that makes it worth developing) of airtightness as a game mechanic is what happens when the area is broken, otherwise it's just another hurdle for the player to jump over so that they can live on their ship without a helmet.

**Comment by tsunamayo** [comment]
@Xerma yes I can understand the appeal but that just too hard to code that way. First to define the volume, then to actually check for airtightness... If implemented that would block based for sure. I also need to code the air escaping system, and that has to be voxel based. 
I am fine with that, on a small fighter you dont really need air a helmet will suffice - but of course I can hear the 100% bricks builder having some issue with that.
Cheers
